### ISSUE TITLE
Woo Express Free Trial to Plan Upgrade: Confusing UX After Plan Expires 

### ISSUE THREAD

**Issue by ash1eygrace** [Automattic/wp-calypso#83120]
### Quick summary

If a user creates a Woo Express free trial site and then upgrades to a paid plan, they face a confusing UX upon plan expiration. Even if they upgraded months prior, they're redirected to a "free trial has ended" page when their plan expires.

**Workaround:** Renew expired WordPress.com plan from: https://wordpress.com/me/purchases
 
### Steps to reproduce

1. Create a free Woo Express trial site: https://woocommerce.com/express/ 
2. Upgrade your site to a paid plan. 
3. Wait for the plan to expire (A8C: Manually change the expiration date of your test site to today).
4. Go to WordPress.com My Site > Choose your site address 

### What you expected to happen

I expect not to see any mention of a free trial site since I had a paid plan. 

### What actually happened

Very Confusing UX:  All site-specific Calypso pages are redirected to https://wordpress.com/plans/my-plan/trial-expired/{site-address}, which says, "Your free trial has ended," even though I upgraded to a paid plan. 

<img width="1636" alt="Pasted image 20231017135441" src="https://github.com/Automattic/wp-calypso/assets/29527450/69e0f8fc-1438-49a9-87ab-b811a870623b">

### Impact

Some (< 50%)

### Available workarounds?

Yes, easy to implement

### Platform (Simple and/or Atomic)

Atomic

### Logs or notes

- 7180601-zd-a8c 
- p1697564370005209-slack-C03TY6J1A
- p1697567836888679-slack-C036E4KFP2L

**Comment by DavidRothstein** [comment]
I'm retitling this since I don't think it's specific to the eCommerce plan.  I can reproduce the exact same screenshot if I upgrade to the Woo Express Performance plan in step 2 instead.  (In that case, the first upgrade option is grayed out because it's a downgrade, and the second is grayed out because I already own it, so I'm stuck.)

I'm also changing the auto-assigned priority here; it does not seem low priority at all. The bug causes most pages in Calypso to be inaccessible (since they redirect to the free trial notice) which prevents customers with active paid plans from administering their site. And this may happen somewhat frequently in the case of monthly plans, which don't auto-renew until the day of expiration and therefore can easily be "expired" even if a renewal hasn't been attempted yet.

I'm not sure if the "your free trial has ended" message should display at all in this case (since their most recent plan wasn't a free trial), but the simplest fix for this bug may be to keep that behavior but just adjust the logic?

Current logic:
- Display the message if the site ever had a free trial and currently has no plan or has an **expired plan**

Proposed new logic:
- Display the message if the site ever had a free trial and currently has no plan or has an **expired free trial plan**

That way the message will continue to appear as soon as their free trial expires (which appears to have been the original intention), but if they own another plan, it won't appear until that plan actually goes away for real.

It looks like the code was originally added in https://github.com/Automattic/wp-calypso/pull/74335.  I'm not sure if Serenity is still the team who maintains this product, or if has moved to someone else.

**Comment by ash1eygrace** [comment]
Hey @DavidRothstein, I definitely agree that there are a lot of nuances here to untangle with this one. 

My focus with this report is more on the separation of .com and Woo flows. Specifically, a WordPress.com eCommerce plan was purchased after having set up a Woo Express Free Trial. I'm assuming we should be preventing these purchases altogether in the checkout, but at the very least if we allow them, we should not throw expired WordPress.com purchases into Woo flows and instead use the .com flow. 

> The bug causes most pages in Calypso to be inaccessible (since they redirect to the free trial notice) which prevents customers with active paid plans from administering their site. 

This aspect has already been reported here: https://github.com/Automattic/wp-calypso/issues/79101

**Comment by DavidRothstein** [comment]
>This aspect has already been reported here: https://github.com/Automattic/wp-calypso/issues/79101

I don't think so -- that issue seems to be about users that have an expired free trial on one site not being able to switch to their **other** sites in Calypso (which is quite bad too).  But the issue here is that you can have an **active** (expired, but still active) paid plan on a single site, but still have the site treated like an expired free trial.

>Specifically, a WordPress.com eCommerce plan was purchased after having set up a Woo Express Free Trial. I'm assuming we should be preventing these purchases altogether in the checkout, but at the very least if we allow them, we should not throw expired WordPress.com purchases into Woo flows and instead use the .com flow.

We need to distinguish between "expired" and "removed" here.
- If the plan is merely expired (but has not yet been removed) then we shouldn't put **any plan** into the above flow, and the fix I suggested above will do that.
- If the plan has gone through its post-expiration grace period and has been removed (so it's truly like a free site again) then I am honestly not sure what the intended behavior is.  The code basically seems to treat sites that started off as Woo Express Free Trial sites as fundamentally different from any other kind of site (forever), regardless of what products they bought later.  I am not sure that is so bad necessarily, but you may be right that unexpected types of purchases (eCommerce plan or otherwise) should be blocked on those sites :shrug:

**Comment by ash1eygrace** [comment]
> But the issue here is that you can have an active (expired, but still active) paid plan on a single site, but still have the site treated like an expired free trial. 

Thank you for clarifying. 

I've updated my original issue to be more in line with this and will open a separate issue about Woo trial sites being able to purchase a WordPress.com plan tomorrow.

**Comment by valterlorran** [comment]
This was fixed by https://github.com/Automattic/wp-calypso/pull/85104
